### PR TITLE
add prometheus metric exporter framework

### PR DESCRIPTION
**Pull request by hangc0276**
### Motivation
KOP doesn't expose prometheus metrics to monitor running stats, which will be hard the debug performance problem.

### Changs
1. add Prometheus metrics provider for kop and use individual port to expose kop metrics
2. Due to BookKeeper Prometheus metric provider include basic metrics stats, such as jvm metrics, memory metrics, cpu metric etc, which will lead those metrics register twice. When run kop with pulsar broker, the prometheus metric thread will start failed.

### CODE REVIEW

**Review by codelipenghui — COMMENTED** [review]
@hangc0276 We can expose metrics through /metrics endpoint of the broker directly, I have add Raw Metrics Provider in Pulsar (https://github.com/apache/pulsar/pull/9021), you can use it to expose additional metrics to the broker metrics endpoint.

**Comment by BewareMyPower** [comment]
The `kafka-impl/conf` won't be used anymore (see #387 ), please edit `docs/configuration.md` or add a new document for exposing metrics to Prometheus under `docs` directory and add a link in `README.md`.

**Comment by hangc0276** [comment]
> @hangc0276 We can expose metrics through /metrics endpoint of the broker directly, I have add Raw Metrics Provider in Pulsar ([apache/pulsar#9021](https://github.com/apache/pulsar/pull/9021)), you can use it to expose additional metrics to the broker metrics endpoint.

@codelipenghui Ok, I update the code and use pulsar raw metrics provider to expose kop metrics to pulsar metrics group. Please take a look. Thanks.

**Comment by BewareMyPower** [comment]
You should fix the code style first. Then please remove `kafka-impl/conf/broker.conf`. It's unused.

Regarding how to enable the monitor for prometheus, we can open a new PR to add the docs.

**Review by jiazhai — APPROVED** [review]

**Comment by jiazhai** [comment]
opened a new issue for the doc. @codelipenghui @BewareMyPower Would you please help review it again?

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopServerStats.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Kop server stats for prometheus metrics.
+ */
+public interface KopServerStats {
+    String CATEGORY_SERVER = "server";
+
+    String SERVER_SCOPE = "kop_server";
+
+    String SERVER_STATUS = "SERVER_STATUS";
+
```
Comment (BewareMyPower) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopServerStats.java:25]:
```suggestion
```

Comment (BewareMyPower) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopServerStats.java:25]: Remove unused variable

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: tests/src/test/java/io/streamnative/pulsar/handlers/kop/MetricsProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        HttpResponse response = httpClient.execute(new HttpGet(metricsEndPoint));
+        InputStream inputStream = response.getEntity().getContent();
+        InputStreamReader isReader = new InputStreamReader(inputStream);
+        BufferedReader reader = new BufferedReader(isReader);
+        StringBuffer sb = new StringBuffer();
+        String str;
+        while ((str = reader.readLine()) != null) {
+            sb.append(str);
+        }
+        Assert.assertTrue(sb.toString().contains("kop_server_HANDLE_PRODUCE_REQUEST"));
```
Comment (BewareMyPower) [inline tests/src/test/java/io/streamnative/pulsar/handlers/kop/MetricsProviderTest.java:124]: Could you add checks for all the metrics you added? Like `kop_server_PRODUCE_ENCODE` and others.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

**Comment by BewareMyPower** [comment]
I've left a few comments, PTAL.

**Comment by hangc0276** [comment]
> I've left a few comments, PTAL.

@BewareMyPower  OK， i will fix it

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**